### ISSUE TITLE
unable to open upcall type 2

### ISSUE THREAD

**Issue by techtonik** [cbodley/ms-nfs41-client#2]
I've installed `v1.0.0/ms-nfs41-client-setup-x86.exe` uploaded in issue #1. But server failed to start with `unable to open upcall type 2` error.

```
C:\Program Files\NFSv4.1 Client>nfsd_debug.exe
parse_cmdlineargs: debug_level 1 ldap is 1
1988: getaddrinfo response flags: 0x0
1988: Family: AF_INET6 IPv6 address ::1
1988: Canonical name: (null)
1988: getdomainname: hostname Box 8
1988: getaddrinfo response flags: 0x0
1988: Family: AF_INET IPv4 address 192.168.50.1
1988: Canonical name: (null)
1988: getdomainname: hostname Box.domain 16
1988: getdomainname: domainname domain 7
1988: NFS41 Daemon starting: version 15494144
1988: Unable to open upcall pipe 2
```

I went to the setup of my network adapter and removed domain from default suffix (it was added by mistake) and the error changed to:

```
C:\Program Files\NFSv4.1 Client>nfsd_debug.exe
parse_cmdlineargs: debug_level 1 ldap is 1
1b68: getaddrinfo response flags: 0x0
1b68: Family: AF_INET6 IPv6 address ::1
1b68: Canonical name: (null)
1b68: getdomainname: hostname Box 8
1b68: getaddrinfo response flags: 0x0
1b68: Family: AF_INET IPv4 address 192.168.50.1
1b68: Canonical name: (null)
1b68: getdomainname: hostname Box 8
1b68: getaddrinfo response flags: 0x0
1b68: Family: AF_INET IPv4 address 192.168.30.1
1b68: Canonical name: (null)
1b68: getdomainname: hostname Box 8
1b68: getaddrinfo response flags: 0x0
1b68: Family: AF_INET IPv4 address 192.168.33.8
1b68: Canonical name: (null)
1b68: getdomainname: hostname Box 8
1b68: getdomainname: unable to get a domain name. Set this machine's domain name
:
System > ComputerName > Change > More > mydomain
```

I am just curious why domain name is so important?

But this unlikely to solve real issue. I also noticed in system log:

```
The following boot-start or system-start driver(s) failed to load: 
nfs41_driver
tvtumon
```

Both logs `nfsddbg.log` and `nfsderr.log` are empty.

**Comment by techtonik** [comment]
I just removed `DFS Replication Service`, maybe it will help.

**Comment by seanlinmt** [comment]
Are you in test mode?

**Comment by techtonik** [comment]
Yes. I likely was in test mode to get access to 8Gb of memory, but that mode doesn't work anymore.

**Comment by JamoDevNich** [comment]
In-case anyone else runs into this issue, here's how I troubleshooted & partially solved this in my scenario...

1. Check whether the driver is running using the command `sc query nfs41_driver`.
    - **If nfs41_driver is running, stop here - this probably won't fix your issue**
    - For good measure, make sure DFS is not running using `sc query Dfsc`. If it is running, using an administrator command prompt run `sc stop Dfsc`, `sc config Dfsc start=disabled` and reboot your machine.

2. Attempt to start nfs41_driver using an administrator command prompt `sc start nfs41_driver`.
    - I got an error complaining about the certificate, so I generated a new self-signed one using these resources:
        - https://stackoverflow.com/a/48817154
        - https://github.com/BAPostma/SigntoolUI
        - Windows 10 SDK (provides signtool.exe)

3. Un-install the old NFS4 driver using the `uninstall.bat` file, and install your newly signed NFS4 driver using `install.bat`

4. Reboot your system with driver signature enforcement disabled.

5. Check whether nfs41_driver is running using `sc query nfs41_driver`.

![Image showing nfs41_driver running](https://i.imgur.com/B0V5J09.png)

So, I mentioned how this was *partially solved...* I did not notice the `unable to open upcall pipe #` error after this, but I have not been able to mount a network drive. The mount command completes with no obvious error, but the mapped drive is absent in Explorer and can't be selected in the Command Prompt.
